src/content/documentation/docs/get-started-postgresql.mdx

```diff
@@ -61,6 +61,72 @@ For an example of using Drizzle ORM with the Neon Serverless driver in a Cloudfl
 
 To use Neon from a serverfull environment, you can use the PostgresJS driver, as described in Neon's **[official nodejs docs](https://neon.tech/docs/guides/node)** — see **[docs](#postgresjs)**.
 
+## Xata
+
+According their **[official website](https://xata.io)**, Xata is a Postgres data platform with a focus on reliability, scalability, and developer experience. The Xata Postgres service is currently in beta, please see the [Xata docs](https://xata.io/docs/postgres) on how to enable it in your account.
+
+Drizzle ORM natively supports both the `xata` driver with `drizzle-orm/xata` package and the **[`postgres`](#postgresjs)** or **[`pg`](#node-postgres)** drivers for accessing a Xata Postgres database.
+
+<Npm>
+drizzle-orm @xata.io/drizzle @xata.io/client
+-D drizzle-kit
+</Npm>
+
+You can use Drizzle with Xata with a HTTP client or a TCP client. The HTTP client doesn't create a persistent connection to the Xata server, while the TCP client does and can be used for better performance with a higher number of requests. The HTTP client is usually recommended from serverless environments like Cloudflare Workers or Vercel Edge Functions. The TCP client is typically used from long-running servers like Express.js or Fastify.
+
+The following example use the Xata generated client, which you obtain by running the [xata init](https://xata.io/docs/getting-started/installation) CLI command.
+
+<Tabs items={['HTTP', 'TCP', 'TCP (pool)']}>
+<Tab>
+```typescript copy filename="index.ts"
+import { drizzle } from '@xata.io/drizzle';
+import { getXataClient } from './xata'; // Generated client
+
+const xata = getXataClient();
+const db = drizzle(xata);
+
+const result = await db.select().from(...);
+```
+</Tab>
+<Tab>
+```typescript copy filename="index.ts"
+import { drizzle } from '@xata.io/drizzle/pg';
+import { getXataClient } from './xata'; // Generated client
+import { Client } from 'pg';
+
+const xata = getXataClient();
+const client = new Client({ connectionString: xata.sql.connectionString });
+const db = drizzle(client);
+```
+</Tab>
+<Tab>
+```typescript copy filename="index.ts"
+import { drizzle } from '@xata.io/drizzle/pg';
+import { getXataClient } from './xata'; // Generated client
+import { Pool } from 'pg';
+
+const xata = getXataClient();
+const pool = new Pool({ connectionString: xata.sql.connectionString, max: 10 });
+const db = drizzle(pool);
+```
+</Tab>
+</Tabs>
+
+If you prefer to not use the generated Xata client, it is also possible to use Xata with the **[`postgres`](#postgresjs)** or **[`pg`](#node-postgres)** drivers, in this case you can copy the connection string from the Settings page of your Xata database. For example:
+
+```typescript copy filename="index.ts"
```

Comment:
I'm on the fence about including this code snippet, since anyway the same instructions are on the same page below. Happy to remove it if you think it's not needed.